visualizer.py

```diff
@@ -0,0 +1,323 @@
+        """ Redraw plots.
+
+        # Args
+            times: The list of timestamps of trace points.
+            traces: A dictionary of all traces to display.
+        """
+        # Update the data store atomically
+        new_datastore = {
+            'time': times / SAMPLE_RATE_HZ,
+        }
+
+        for trace in traces.values():
+            new_datastore[trace.label] = trace.values * trace.scale
+
+        self._data_store = new_datastore
+
+        figure = bokeh.plotting.figure(output_backend="webgl", sizing_mode="stretch_both")
+
+        # Update traces
+        palette = bokeh.palettes.d3['Category10'][len(traces.keys())]
+        for (trace, color) in zip(traces.values(), palette):
+            figure.circle(x='time', y=trace.label, source=self._data_store, color=color,
+                          legend_label=trace.label)
+
+        figure.legend.location = 'top_left'
+        figure.legend.click_policy = 'hide'
+
+        self.layout.children[0] = figure
+
+
+    def capture(self, duration: float = None):
+        """ Initialize a data capture from Stabilizer's datastream.
+
+        Args:
+            duration: The duration to capture for.
+        """
+        if duration is None:
+            duration = float(self._capture_duration_input.value)
+
+        self.receiver.capture(duration)
+
+        # TODO: Force the trigger for now. Once we allow triggered waveforms, this force will be
+        # removed.
+        self.receiver.force_trigger()
+
+        # Start periodicly reading trigger state.
+        self._callback = self.doc.add_periodic_callback(self.update,
+                                                        period_milliseconds=1000/60)
+
+
+async def configure_streaming(prefix, broker, port):
+    """ Set up Stabilizer streamining to the desired endpoint.
+
+    Args:
+        port: The port to direct the stream to.
+        broker: The IP address of the broker.
+        prefix: The miniconf prefix of Stabilizer.
+    """
+    _, local_ip = _get_ip(broker)
+    interface = await Miniconf.create(prefix, broker)
+
+    # Configure the stream
+    print(f'Configuring stream to target {".".join(map(str, local_ip))}:{port}')
+    print('')
+    await interface.command('stream_target', {'ip': local_ip, 'port': port}, retain=False)
+
+
+async def main():
+    """ Main entry point. """
+    logging.basicConfig(level=logging.INFO)
+    parser = argparse.ArgumentParser(description='Stabilizer livestream viewer')
+    parser.add_argument('--prefix', type=str, help='The MQTT topic prefix of the target. '
+                                                   'If provided, streaming will be configured')
+    parser.add_argument('--broker', '-b', default='mqtt', type=str,
+                        help='The MQTT broker address')
+    parser.add_argument('--port', '-p', default=2000, type=int,
+                        help='The UDP port to use for streaming')
```

Comment:
The README (or help) should contain a hint that the firewall on the backend machine may need to be opened up explicitly to receive the stream data (e.g. `ufw allow 9293/udp` for ufw users).